Apps/CocoaPods-FCM/src/View/DashboardView.swift

```diff
@@ -1,8 +1,102 @@
 import CioDataPipelines
+import CioMessagingInApp
 import SwiftUI
+import UIKit
 import UserNotifications
 
+// ObservableObject that tracks the height of a subview
+class HeightTracker: ObservableObject {
+    @Published var subviewHeight: CGFloat = 0
+}
+
+// Used for associating the height tracker with InAppMessageView
+private var heightTrackerKey: UInt8 = 0
+private var maxHeight: CGFloat = 0
+
+extension InAppMessageView {
+    var heightTracker: HeightTracker? {
+        get {
+            objc_getAssociatedObject(self, &heightTrackerKey) as? HeightTracker
+        }
+        set {
+            objc_setAssociatedObject(self, &heightTrackerKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    // Tracks and update the height
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+
+        print("Height of tracker \(String(describing: heightTracker?.subviewHeight))")
+
+        // Update maxHeight and notify heightTracker if the frame height changes
+        // The height fluctuates, increasing and then decreasing.
+        // To prevent this behavior, we constrain it to the maximum height achieved by the subview.
+        if frame.height >= maxHeight {
+            maxHeight = frame.height
+            heightTracker?.subviewHeight = frame.height
+        }
+    }
```

Comment:
why is this code needed?

Reply:
[In the demo video](https://linear.app/customerio/issue/MBL-402/display-an-inline-message-in-a-swiftui-app#comment-8a1582f9), I noticed that when the inline View renders, the height of the View instantly jumps from 0 to N. The UIKit view has an animation that animates every time the height is changed, including when the inline View is dismissed. 

Animations matching the UIKit View is part of the acceptance criteria of the ticket so we'll need to get that working.